extensions/widgets/iconpicker/iconpicker.lcb

```diff
@@ -177,13 +187,24 @@ private handler drawIcon(in pIcon as Integer)
 		
 	constrainPathToRect(mIconRects[pIcon], tPath)
 	
-	if mSelectedElement is not nothing and tIconName is mSelectedElement then
-		set the paint of this canvas to getPaint("icon", "selected")
-		fill rectangle path of indexToSectorRect(pIcon) on this canvas
-	end if
-	
 	set the paint of this canvas to solid paint with color kIconColor
 	fill tPath on this canvas
+	
+	if mShowNames then
+		variable tTextRect as Rectangle
+		put mIconRects[pIcon] into tTextRect
+		set the height of tTextRect to kNameHeight
+		set the top of tTextRect to the top of tTextRect - kNameHeight
+		
+		variable tLabel as String
+		put tIconName into tLabel
+	
+		if the number of chars in tIconName > kMaxIconNameChars then
+			put char 1 to kMaxIconNameChars of tIconName & "\u{2026}" into tLabel
```

Comment:
Total pedantry: surely this should be `char 1 to (kMaxIconNameChars - 1)`, because the ellipsis takes up a char?